synapse/rest/client/shared_rooms.py

```diff
@@ -43,16 +43,17 @@ def __init__(self, hs: "HomeServer"):
         self.auth = hs.get_auth()
         self.store = hs.get_datastores().main
         self.user_directory_active = hs.config.server.update_user_directory
+        self.hs = hs
 
     async def on_GET(
         self, request: SynapseRequest, user_id: str
     ) -> Tuple[int, JsonDict]:
 
-        if not self.user_directory_active:
+        if not self.hs.config.userdirectory.user_directory_search_enabled:
```

Comment:
Can we make this a variable rather than saving `self.hs`? It's a bit of a smell to have `self.hs` lying around if you can avoid it.